integration/apps/parres/parres.py

```diff
@@ -0,0 +1,296 @@
+
+        if not ( ( self._cores_per_node // self.get_cpu_per_rank() == 1 ) and gpus_per_node == 1):           
+            raise RunTimeError('Can currently only handle 1 ranks per node and 1 gpus per node')
+                
+        benchmark_dir = os.path.dirname(os.path.abspath(__file__))
+
+        if not parres_init_setup is None:
+            if not os.path.isfile(parres_init_setup):
+                if (os.path.isfile(os.path.join(benchmark_dir, parres_init_setup))):
+                    parres_init_setup = os.path.join(benchmark_dir, parres_init_setup)
+                else:
+                    raise runtimeError("Parres-init-setup file not found:" + parres_init_setup)
+            os.chmod(parres_init_setup, 0o755)
+            subprocess.call(parres_init_setup, shell=True)
+
+        if not parres_exp_setup is None:
+            if not os.path.isfile(parres_exp_setup):
+                if (os.path.isfile(os.path.join(benchmark_dir, parres_exp_setup))):
+                    parres_exp_setup = os.path.join(benchmark_dir, parres_exp_setup)
+                else:
+                    raise runtimeError("Parres-exp-setup file not found:" + parres_exp_setup)
+            
+        if not parres_teardown is None:
+            if not os.path.isfile(parres_teardown):
+                if (os.path.isfile(os.path.join(benchmark_dir, parres_teardown))):
+                    parres_teardown = os.path.join(benchmark_dir, parres_teardown)
+                else:
+                    raise runtimeError("Parres-teardown file not found:" + parres_teardown)
+                
+        self._parres_exp_setup = parres_exp_setup
+        self._parres_teardown = parres_teardown
+
+        binary_name = 'Kernels/Cxx11/nstream-mpi-cublas'
+        if parres_args is None:
+            params = ["1","1000000000"]
+        else:
+            params=parres_args.split()
+
+        self.app_params = " ".join(params)
+
+        self.exe_path = os.path.join(benchmark_dir, binary_name)
+
+
+    def get_rank_per_node(self):
+        return self._cores_per_node // self.get_cpu_per_rank()
+
+    def get_cpu_per_rank(self):
+        return self._cores_per_rank
+
+    def get_bash_exec_path(self):
+        return self.exe_path
+
+    def get_bash_exec_args(self):
+        return self.app_params
+
+    def get_bash_setup_commands(self):
+        setup_commands = textwrap.dedent('''
+        ulimit -s unlimited
+        export I_MPI_FABRICS=shm
+        ''')
+
+        return setup_commands
+
+    def trial_setup(self, run_id, output_dir):
+        if not self._parres_exp_setup is None:
+            os.chmod(self._parres_exp_setup, 0o755)
+            subprocess.call(self._parres_exp_setup, shell=True)
+
+        
+    
+    def experiment_teardown(self, output_dir):
+        if not self._parres_teardown is None:
+            os.chmod(self._parres_teardown, 0o755)
+            subprocess.call(self._parres_teardown, shell=True)
+
+
```

Comment:
```suggestion
```suggestion
def parse_fom(self, log_path):
        result = None
        key = 'Rate (MB/s): '
        with open(log_path) as fid:
            for line in fid.readlines():
                if key in line:
                    result = line.split(': ')[1]
                    result = float(result.split(' ')[0])
                    break
        return result
```
```

Reply:
Added FOM code.